### PR TITLE
[Merged by Bors] - Fix zookeeper quorum and port configuration

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Fixes https://github.com/stackabletech/hbase-operator/issues/354 and https://github.com/stackabletech/hbase-operator/issues/355

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/zookeeper.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Contains the information as exposed by the Zookeeper/Znode discovery CM (should work with both)
+pub struct ZookeeperConnectionInformation {
+    /// E.g. `simple-zk-server-default-0.simple-zk-server-default.default.svc.cluster.local:2282,simple-zk-server-default-1.simple-zk-server-default.default.svc.cluster.local:2282,simple-zk-server-default-2.simple-zk-server-default.default.svc.cluster.local:2282`
+    pub hosts: String,
+    /// E.g. `/znode-123` in case of ZNode discovery CM or `/` in case of Zookeeper discovery CM directly.
+    pub chroot: String,
+    /// E.g. 2282 for tls secured Zookeeper
+    pub port: u16,
```
Comment (maltesander) [inline rust/operator-binary/src/zookeeper.rs:48]: Dont need to be public.

**Review by maltesander — APPROVED** [review]
LGTM! Just minor things

File: rust/operator-binary/src/zookeeper.rs (hunk 1 above)
Comment (sbernauer) [reply rust/operator-binary/src/zookeeper.rs:48]: Not any more after refactoring. nice!

**Comment by sbernauer** [comment]
Started test https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hbase-operator-it-custom/90/

**Comment by sbernauer** [comment]
bors r+
Test passed

**Pull request closed without merging**